File: src/main/java/projetFinal/entity/Plat.java
package projetFinal.entity;

import java.util.HashSet;
import java.util.Set;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Lob;
import javax.persistence.OneToMany;
import javax.persistence.SequenceGenerator;
import javax.persistence.Table;

@Entity
@Table(name = "plat")
@SequenceGenerator(name = "seqPlat", sequenceName = "seq_plat", allocationSize = 1)
public class Plat {
	@Id
	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "seqPlat")
	@Column(name = "plat_id")
	private Long id;
	@Column(name = "plat_nom", nullable = false, length = 200)
	private String nom;
	@Lob
	@Column(name = "plat_description")
	private String description;
	@Lob
	@Column(name = "plat_photo")
	private byte[] photo;
	@OneToMany(mappedBy = "id.plat")
	private Set<LigneCommande> lignesCommande = new HashSet<LigneCommande>();
	@OneToMany(mappedBy = "id.plat")
	private Set<LigneCarte> lignesCarte;
	@Column(name = "ligne_carte_prix")
	private Float prix;

	public Plat() {

	}

	public Plat(String nom, String description) {
		super();
		this.nom = nom;
		this.description = description;
	}

	public Plat(String nom, String description, byte[] photo) {
		this.nom = nom;
		this.description = description;
		this.photo = photo;
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getNom() {
		return nom;
	}

	public void setNom(String nom) {
		this.nom = nom;
	}

	public Float getPrix() {
		return prix;
	}

	public void setPrix(Float prix) {
		this.prix = prix;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public byte[] getPhoto() {
		return photo;
	}

	public void setPhoto(byte[] photo) {
		this.photo = photo;
	}

	public Set<LigneCommande> getLignesCommandes() {
		return lignesCommande;
	}

	public void setLignesCommandes(Set<LigneCommande> lignesCommandes) {
		this.lignesCommande = lignesCommandes;
	}

	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + ((id == null) ? 0 : id.hashCode());
		return result;
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Plat other = (Plat) obj;
		if (id == null) {
			if (other.id != null)
				return false;
		} else if (!id.equals(other.id))
			return false;
		return true;
	}

}

File: HashiAngular/src/app/model/plat.ts
import { Byte } from '@angular/compiler/src/util';

export class Plat {
  constructor(
    private _id?: number | undefined,
    private _nom?: string | undefined,
    private _description?: string | undefined,
    private _photo?: Byte[] | undefined,
    private _prix?: number | undefined
  ) {}

  /**
   * Getter prix
   * @return {number}
   */
  public get prix(): number | undefined {
    return this._prix;
  }

  /**
   * Setter prix
   * @param {number} value
   */
  public set prix(value: number | undefined) {
    this._prix = value;
  }

  /**
   * Getter photo
   * @return {Byte[]}
   */
  public get photo(): Byte[] | undefined {
    return this._photo;
  }

  /**
   * Setter photo
   * @param {Byte[]} value
   */
  public set photo(value: Byte[] | undefined) {
    this._photo = value;
  }

  /**
   * Getter description
   * @return {string}
   */
  public get description(): string | undefined {
    return this._description;
  }

  /**
   * Setter description
   * @param {string} value
   */
  public set description(value: string | undefined) {
    this._description = value;
  }

  /**
   * Getter nom
   * @return {string}
   */
  public get nom(): string | undefined {
    return this._nom;
  }

  /**
   * Setter nom
   * @param {string} value
   */
  public set nom(value: string | undefined) {
    this._nom = value;
  }

  /**
   * Getter id
   * @return {number}
   */
  public get id(): number | undefined {
    return this._id;
  }

  /**
   * Setter id
   * @param {number} value
   */
  public set id(value: number | undefined) {
    this._id = value;
  }
}
